src/main/java/com/example/roster10/StaffService.java

```diff
@@ -5,21 +5,49 @@
 import java.time.LocalDate;
 import java.util.Optional;
 
-@Service
-public class StaffService {
-    private final StaffMapper staffMapper;
-
-    public StaffService(StaffMapper staffMapper) {
-        this.staffMapper = staffMapper;
-    }
-
-    public Staff insert(String name, LocalDate dateOfBirth, String nearestStation) {
-        Staff staff = Staff.createStaff(name, dateOfBirth, nearestStation);
-        staffMapper.insert(staff);
-        return staff;
-    }
-    public Staff findStaff(int id) {
-        Optional<Staff> roster = this.staffMapper.findById(id);
-        return roster.orElseThrow(() -> new StaffNotFoundException("Staff with id " + id + " not found"));
-    }
+
+
+
+    @Service
+    public class StaffService {
+        private final StaffMapper staffMapper;
+
+        public StaffService(StaffMapper staffMapper) {
+            this.staffMapper = staffMapper;
+        }
+
+        public Staff insert(String name, LocalDate dateOfBirth, String nearestStation) {
+            Staff staff = Staff.createStaff(name, dateOfBirth, nearestStation);
+            staffMapper.insert(staff);
+            return staff;
+        }
+
+        public Staff findStaff(int id) {
+            Optional<Staff> staff = this.staffMapper.findById(id);
+            return staff.orElseThrow(() -> new StaffNotFoundException("Staff with id " + id + " not found"));
+        }
+        public void updateStaff(int id, String name, LocalDate dateOfBirth, String nearestStation) {
+            // まず、スタッフを検索する
+            Optional<Staff> staffOptional = staffMapper.findById(id);
+            // スタッフが見つからない場合は例外をスローする
+            Staff existingStaff = staffOptional.orElseThrow(() -> new StaffNotFoundException("Staff with id " + id + " not found"));
+
+            // existingStaffを更新する
+//        existingStaff.setName(name);
+//        existingStaff.setDateOfBirth(dateOfBirth);
+//        existingStaff.setNearestStation(nearestStation);
+            // リクエストで送られてきた値が null でない場合にのみ更新する
+            if (name != null) {
```

Comment:
余談ですが、Objectクラスに以下のようなメソッドも用意されています。
好みですが、こちらの方が私は可読性が高いと思ってます。(参考までに🙏)
```suggestion
            if (Object.nonNull(name)) {
```

Reply:
調べて書き替えてみました。動作確認OKです。ご確認お願い致します😌
![image](https://github.com/hiro903/Kadai10/assets/145466271/fbe48eac-afdb-4939-bd2b-ad0367ec5019)
![image](https://github.com/hiro903/Kadai10/assets/145466271/c9e5b8c2-9bd0-46c1-8807-d4b4de768f97)
![image](https://github.com/hiro903/Kadai10/assets/145466271/d9de64ac-75fc-4efd-b34c-4f36ce7283fe)
![image](https://github.com/hiro903/Kadai10/assets/145466271/c5fd2c66-6f0e-4533-b51a-01e2ccc8c9dc)